sklearn/impute/_base.py

```diff
@@ -628,7 +645,15 @@ def _fit(self, X, y=None):
             The imputer mask of the original data.
 
         """
-        X = self._validate_input(X)
+        self._precomputed = False
+        if sparse.issparse(X) or isinstance(X, np.ndarray):
+            if X.dtype == 'bool' and self.missing_values:
+                self._precomputed = True
+
+        # Need not validate X again as it would have already been validated
```

Comment:
I would change this comment of location and shorten it

```python
if not self._precomputed_mask:
    # don't validate X since it was validated by the Imputer.
```

Reply:
However, I am wondering if we should not validate `X` if the a user is using the `MissingIndicator` alone?

Reply:
To overcome this, we were also checking whether or not `self.missing_values==True`, when setting `precomputed=True`

Reply:
Hmm..but then again, I wonder if the user set `X` as `boolean` and `self.missing_values=True`, it would crumble.
I'll check for some condition to make sure that doesn't happen.